pages/upgrade-guide/4.0.md

```diff
@@ -0,0 +1,186 @@
+
+The recommendation for migration is to be explicit about which classes should be toggled.
+
+### Core
+
+#### Breaking change: Deprecated APIs removed
+
+The following APIs were deprecated in previous versions and have been removed in jQuery 4.0:
+
+* `jQuery.camelCase`
+	This function was never documented and was only used internally. It was deprecated in jQuery 3.0.
+
+* `jQuery.cssProps`
+	This object was used to map CSS properties to their vendor-prefixed versions, which is no longer necessary in jQuery 4.0's supported browsers.
+
+* `jQuery.isArray`
+* `jQuery.isFunction`
+* `jQuery.isNumeric`
+* `jQuery.isWindow`
+* `jQuery.fx.interval`
+* `jQuery.parseJSON`
+* `jQuery.nodeName`
+* `jQuery.now`
+* `jQuery.trim`
+* `jQuery.type`
+
+#### Breaking change: Remove undocumented Array methods
+
+jQuery 4.0 removes `push`, `sort`, and `splice` from the jQuery prototype. These methods were copies of the methods of the same names on the native Array prototype, but were never documented and were not intended for public use. Some plugins may have used them, so we are still marking this as a breaking change.
+
+#### Slim build breaking change: Remove `callbacks` and `deferred` modules
+
+The slim build of jQuery 4.0 no longer includes the `callbacks` and `deferred` modules. Use the full build if you need these modules, or use native Promise objects.
+
+#### Breaking change: Remove undocumented `root` parameter of `jQuery()`
+
+The `root` parameter of `jQuery()` was never documented and was only used internally. Also, it has not been necessary since jQuery 1.9 when `jQuery.sub()` was removed.
+
+#### Important fix: Add `exports` to package.json
+
+jQuery exports multiple builds that can be used in different environments. This includes the default build as both a universal module (UMD) and an ESM module, the slim build in UMD and ESM, and a factory build for running jQuery with DOM emulators like JSDOM. The `exports` field in `package.json` is used to specify which build is used in different environments. This change is not expected to affect existing code.
+
+### CSS
+
+#### Breaking change: `px` no longer automatically added to most unitless values
+
+jQuery 4.0 switches strategies when it comes to setting CSS properties that require units. Previously, jQuery would automatically add `px` to unitless values, with some exceptions. This behavior has been removed in jQuery 4.0.
+
+Instead, jQuery will now only add `px` to a limited set of properties that are known to require it. This change may affect existing code that relies on the old behavior.
+
+### Data
```

Comment:
I'd mention that the `.data()` setter used to convert `-ms-x` to `msX` and now to `MsX`. This is a result of separating `camelCase` logic between `data` & `css` modules.

Reply:
I think that's too niche. Also, with the removal of support for IE10 and below, I'd rather encourage users not to use `-ms` at all, unless there are cases where it's still applicable.

Reply:
OK, fair enough. I think we don't even handle that in Migrate, although maybe that part we should fix.

Reply:
I created https://github.com/jquery/jquery-migrate/issues/579 to consider a Migrate patch for this.